plugin-server/src/main/ingestion-queues/session-recordings-consumer.ts

```diff
@@ -42,386 +37,271 @@ export const startSessionRecordingEventsConsumer = async ({
         Additionally, we process `$performance_event` events which are closely
         tied to session recording events.
 
-        NOTE: it may be safer to also separate processing of
-        `$performance_event` but for now we'll keep it in the same consumer.
+        We use the node-rdkafka library for handling consumption and production
+        from Kafka. Note that this is different from the other consumers as this
+        is a test bed for consumer improvements, which should be ported to the
+        other consumers.
+
+        We consume batches of messages, process these to completion, including
+        getting acknowledgements that the messages have been pushed to Kafka,
+        then commit the offsets of the messages we have processed. We do this
+        instead of going completely stream happy just to keep the complexity
+        low. We may well move this ingester to a different framework
+        specifically for stream processing so no need to put too much work into
+        this.
     */
 
-    // We use our own producer as we want to ensure that we don't have out of
-    // band calls to the `flush` method via the KAFKA_FLUSH_FREQUENCY_MS option.
-    // This ensures that we can handle Kafka Producer errors within the body of
-    // the Kafka consumer handler.
     const groupId = 'session-recordings'
     const sessionTimeout = 30000
+    const fetchBatchSize = 500
```

Comment:
we'll want to make that configurable in a follow-up PR, and measure impact.
Fine to leave it hardcoded for now.